src/main/java/TaskList.java

```diff
@@ -0,0 +1,196 @@
+        tasks.add(deadline);
+
+        Ui.printLine();
+        Ui.indent("Acknowledged. I have added: ");
+        Ui.doubleIndent( deadline.toString());
+        Ui.printTaskCount(this);
+        Ui.printLine();
+    }
+
+    /**
+     * Adds an Event object to the task list
+     * @param args A String in the format of: "<desc> /at <date>"
+     */
+    public void addEvent(String args) {
+        String[] descAndAt = args.split(" /at ");
+        Event event = new Event(descAndAt[0], descAndAt[1]);
+        tasks.add(event);
+
+        Ui.printLine();
+        Ui.indent("Acknowledged. I have added: ");
+        Ui.doubleIndent(event.toString());
+        Ui.printTaskCount(this);
+        Ui.printLine();
+    }
+
+    /**
+     * Add a Todo object to the task list
+     * @param args A String in the format of: "<desc>"
+     */
+    public void addTodo(String args) {
+        Todo todo = new Todo(args);
+        tasks.add(todo);
+        Ui.printLine();
+        Ui.indent("Acknowledged. I have added: ");
+        Ui.doubleIndent(todo.toString());
+        Ui.printTaskCount(this);
+        Ui.printLine();
+    }
+
+    private void add(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * The number of tasks in the TaskList
+     * @return The number of tasks in the TaskList
+     */
+    public int size() {
+        return tasks.size();
+    }
+
+    /**
+     * The Task at a the index specified by the param
+     * @param idx 0-based indexing to get the Task object
+     * @return
+     */
+    public Task get(int idx) {
+        return tasks.get(idx);
+    }
+
+    /**
+     * Mark a Task (specified by the index) as done
+     * @param taskNo A 1-based indexing of the position of the Task in the list
+     * @throws InvalidIndexException
+     */
+    public void markTaskAsDone(int taskNo) throws InvalidIndexException {
+        if (taskNo < 1 || taskNo > tasks.size()) {
+            throw new InvalidIndexException();
+        }
+        Task task = tasks.get(taskNo - 1); // The user starts counting from 1
+        task.markAsDone();
+
+        Ui.printLine();
+        Ui.indent("Excellent! You have completed this task: ");
+        Ui.doubleIndent(task.toString());
+        Ui.printLine();
+    }
+
+    /**
+     * Delete a Task (specified by the index)
+     * @param taskNo A 1-based indexing of the position of the Task in the list
+     * @throws InvalidIndexException
+     */
+    public void deleteTask(int taskNo) throws InvalidIndexException {
+        if (taskNo < 1 || taskNo > tasks.size()) {
+            throw new InvalidIndexException();
+        }
+        Task task = tasks.get(taskNo - 1); // The user starts counting from 1
+        tasks.remove(taskNo - 1);
+
+        Ui.printLine();
+        Ui.indent("Understood. I have now removed this task: ");
+        Ui.doubleIndent(task.toString());
+        Ui.printTaskCount(this);
+        Ui.printLine();
```

Comment:
as mentioned earlier, you could abstract this portion to a simple Ui.print() function and reduce your code length.